mcstatus/status_response.py

```diff
@@ -0,0 +1,326 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from typing import Any, Dict, List, Optional, Union
+
+STYLE_MAP = {
+    "bold": "l",
+    "italic": "o",
+    "underlined": "n",
+    "obfuscated": "k",
+    "color": {
+        "dark_red": "4",
+        "red": "c",
+        "gold": "6",
+        "yellow": "e",
+        "dark_green": "2",
+        "green": "a",
+        "aqua": "b",
+        "dark_aqua": "3",
+        "dark_blue": "1",
+        "blue": "9",
+        "light_purple": "d",
+        "dark_purple": "5",
+        "white": "f",
+        "gray": "7",
+        "dark_gray": "8",
+        "black": "0",
+    },
+}
+
+
+@dataclass
+class AbstractDataclass(ABC):
+    """Abstract base class for all dataclasses.
+
+    We can't use just `ABC` because we need raise TypeError if we try to
+    instantiate an abstract dataclass. Class which inherit from `ABC` raises
+    it only if it has abstract methods.
+    """
+
+    def __new__(cls, *args, **kwargs):
+        if cls == AbstractDataclass or cls.__bases__[0] == AbstractDataclass:
```

Comment:
This should be using an `is` check for comparison of exact objects, not `==`

```suggestion
        if cls is AbstractDataclass or cls.__bases__[0] is AbstractDataclass:
```

Reply:
Changed in https://github.com/py-mine/mcstatus/pull/306/commits/126cf7a0e6b2f34c6a8eec4a70dcb37d8dd6daf5.